src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -1239,7 +1239,9 @@ public void execute() throws MojoExecutionException, CompilationFailureException
             }
         }
 
-        if (outputTimestamp != null && (outputTimestamp.length() > 1 || Character.isDigit(outputTimestamp.charAt(0)))) {
+        if (outputTimestamp != null
+                && !outputTimestamp.isEmpty()
```

Comment:
I'd like to understand how the value can be empty: `<foo/>`? This basically can happen in any plugin using this value, no?

Reply:
Yes, I think it would be set by another plugin. The project I was working on has a lot of plugins, so I would not know which plugin exactly sets this value. When I encountered this issue, the value was not set in the projects pom, but it was a non-null empty string when this line was run, indicating it was set somewhere in the compilation process.

Reply:
While I consider your fix as reasonable, I'd prefer to see the actual cause to be addressed as well...

Reply:
I will look into it.

Reply:
Hi I looked a bit further into this. 

In the (internal) project I was working on was using the jgitver maven extension. This extension has a field named COMMIT_ISO_TIMESTAMP which contains the timestamp of the commit. If the repository does not have any commits, this timestamp is empty. The project I was working on was mapping this timestamp to the project.build.outputTimestamp to this maven commit timestamp in the parent pom, which I missed before.

This caused the above mentioned compiler error.

Reply:
Have you raised an upstream issue?

Reply:
I'll give the maintainers some time to respond.

Reply:
Hi,

On July 2th this MR has been approved by gnodet. Are we waiting for someone elses approval?

Regards,

Thijs Rijpert